cookbook/security/custom_authentication_provider.rst

```diff
@@ -218,12 +219,12 @@ the ``PasswordDigest`` header value matches with the user's password.
     class WsseProvider implements AuthenticationProviderInterface
     {
         private $userProvider;
-        private $cacheDir;
+        private $cachePool;
 
-        public function __construct(UserProviderInterface $userProvider, $cacheDir)
+        public function __construct(UserProviderInterface $userProvider, CacheItemPoolInterface $cachePool)
```

Comment:
Imo we should type hint against the [AdapterInterface](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/AdapterInterface.php) from the Cache component. This way we can inject cache adapters defined through the `framework.cache` config (see below).